dask/utils.py

```diff
@@ -935,9 +936,15 @@ def ensure_unicode(s) -> str:
     """
     if isinstance(s, str):
         return s
-    if hasattr(s, "decode"):
+    elif hasattr(s, "decode"):
         return s.decode()
-    raise TypeError(f"Object {s} is neither a str object nor has an decode method")
+    else:
+        try:
+            return codecs.decode(s)
```

Comment:
Since we're broadening the types of objects `ensure_unicode` can handle, can we update the docstring? `Turn string or bytes to string` seems to not quite fit anymore

Reply:
Yeah was thinking about that. Have tried pushing a change for this and `ensure_bytes` (which had a similar issue). Please let me know what you think 🙂